### PR TITLE
En_Fish

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Fish/z_en_fish.c [hunk 1]
```diff
@@ -42,87 +81,875 @@ const ActorInit En_Fish_InitVars = {
...
+    this->unk_242 = 400;
+    this->unk_25C = this->unk_260 * 0.1f;
+    func_8091D728(this);
+    this->actor.draw = NULL;
+}
+
+Actor* func_8091D944(EnFish* this, GlobalContext* globalCtx) {
+    f32 dist;
+    Actor* retActor = NULL;
+    f32 minDist = FLT_MAX;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Fish/z_en_fish.c:148]:
```suggestion
    f32 distSq;
    Actor* retActor = NULL;
    f32 minDistSq = FLT_MAX;
```
These are square of distances, as opposed to distances

File: src/overlays/actors/ovl_En_Fish/z_en_fish.c [hunk 2]
```diff
@@ -42,87 +81,875 @@ const ActorInit En_Fish_InitVars = {
...
+    s32 pad;
+    Player* player = GET_PLAYER(globalCtx);
+    Vec3f sp1C;
+
+    if (this->actor.xzDistToPlayer < 50.0f) {
+        sp1C.x = (Math_SinS(BINANG_ROT180(this->actor.yawTowardsPlayer)) * 30.0f) + player->actor.world.pos.x;
+        sp1C.y = player->actor.world.pos.y;
+        sp1C.z = (Math_CosS(BINANG_ROT180(this->actor.yawTowardsPlayer)) * 30.0f) + player->actor.world.pos.z;
+
+        if (func_8091D630(&sp1C, &this->actor.world.pos) <= 900.0f) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Fish/z_en_fish.c:252]:
```suggestion
        if (func_8091D630(&sp1C, &this->actor.world.pos) <= SQ(30.0f)) {
```

File: src/overlays/actors/ovl_En_Fish/z_en_fish.c [hunk 3]
```diff
@@ -42,87 +81,875 @@ const ActorInit En_Fish_InitVars = {
...
+}
+
+void func_8091E128(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnFish* this = THIS;
+
+    func_8091DD48(this);
+    Math_SmoothStepToF(&thisx->speedXZ, 1.8f, 0.08f, 0.4f, 0.0f);
+
+    if ((func_8091D630(&thisx->world.pos, &thisx->home.pos) > 6400.0f) || (this->unk_240 < 4)) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Fish/z_en_fish.c:334]:
```suggestion
    if ((func_8091D630(&thisx->world.pos, &thisx->home.pos) > SQ(80.0f)) || (this->unk_240 < 4)) {
```

File: src/overlays/actors/ovl_En_Fish/z_en_fish.c [hunk 4]
```diff
@@ -42,87 +81,875 @@ const ActorInit En_Fish_InitVars = {
...
+    GlobalContext* globalCtx = globalCtx2;
+    EnFish* this = THIS;
+    s32 sp3C = thisx->xzDistToPlayer < 60.0f;
+    s32 sp38 = this->unk_276 > 0;
+    s32 pad;
+
+    func_8091DD48(this);
+    Math_SmoothStepToF(&thisx->speedXZ, 4.2f, 0.08f, 1.4f, 0.0f);
+
+    if (func_8091D630(&thisx->world.pos, &thisx->home.pos) > 25600.0f) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Fish/z_en_fish.c:385]:
```suggestion
    if (func_8091D630(&thisx->world.pos, &thisx->home.pos) > SQ(160.0f)) {
```

File: src/overlays/actors/ovl_En_Fish/z_en_fish.c [hunk 5]
```diff
@@ -42,87 +81,875 @@ const ActorInit En_Fish_InitVars = {
...
+    Player* player = GET_PLAYER(globalCtx);
+    s32 pad;
+    Vec3f sp38;
+    s32 pad2;
+    s16 sp32;
+
+    func_8091DD48(this);
+    Math_SmoothStepToF(&thisx->speedXZ, 1.8f, 0.1f, 0.5f, 0.0f);
+
+    if (func_8091D630(&thisx->world.pos, &thisx->home.pos) > 6400.0f) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Fish/z_en_fish.c:460]:
```suggestion
    if (func_8091D630(&thisx->world.pos, &thisx->home.pos) > SQ(80.0f)) {
```

File: src/overlays/actors/ovl_En_Fish/z_en_fish.c [hunk 6]
```diff
@@ -42,87 +81,875 @@ const ActorInit En_Fish_InitVars = {
...
+    } else {
+        sp3C = &D_8091FADC;
+    }
+
+    func_8091DD48(this);
+    Math_StepToF(&this->actor.world.pos.y, this->actor.home.pos.y, 2.0f);
+    Math_SmoothStepToF(&this->actor.speedXZ, sp3C->x, sp3C->y, sp3C->z, 0.0f);
+    this->unk_24C = 0.0f;
+
+    if (func_8091D630(&this->actor.world.pos, &this->actor.home.pos) > 225.0f) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Fish/z_en_fish.c:809]:
```suggestion
    if (func_8091D630(&this->actor.world.pos, &this->actor.home.pos) > SQ(15.0f)) {
```

File: src/overlays/actors/ovl_En_Fish/z_en_fish.c [hunk 7]
```diff
@@ -42,87 +81,875 @@ const ActorInit En_Fish_InitVars = {
...
+        func_8091E2E0(this);
+    }
+}
+
+void func_8091E070(EnFish* this) {
+    static s16 D_8091FAD4[] = {
+        20,
+        30,
+        40,
+    };
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Fish/z_en_fish.c:303]:
```suggestion
    static s16 D_8091FAD4[] = {
        20,
        30,
        40
    };
```
You can remove the comma when they format into a vertical column like this

**Review by engineer124 — COMMENTED** [review]
Good work. A few comments

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_En_Fish/z_en_fish.c [hunk 8]
```diff
@@ -42,87 +80,874 @@ const ActorInit En_Fish_InitVars = {
...
+    if (this->actor.params == ENFISH_1) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    if ((this->actor.child != NULL) && (this->actor.child->update == NULL) && (thisx != this->actor.child)) {
+        this->actor.child = NULL;
+    }
+
+    if ((this->unkFunc == NULL) || (this->unkFunc(&this->actor, globalCtx), (this->actor.update != 0))) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Fish/z_en_fish.c:902]:
```suggestion
    if ((this->unkFunc == NULL) || (this->unkFunc(&this->actor, globalCtx), (this->actor.update != NULL))) {
```

**Review by hensldm — APPROVED** [review]
Just one small cosmetic change.

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**